### PR TITLE
public static PRISM_TYPE and Nested Annotations Support/JDK 11

### PR DESCRIPTION
**Pull request by SentryMan**
- Now can use the generated Prism to get the annotation fully qualified type as a string, 
- Now supports inner annotations e.g.

```
public @interface OpenAPIDefinition {
// will auto-generate prisms for each of these annotation types
    Info info() default @Info;

    Tag[] tags() default {};
}
```

this will generate:

```

/** A Prism representing an {@code @io.swagger.v3.oas.annotations.OpenAPIDefinition} annotation. 
  */ 
public class OpenAPIDefinitionPrism {

    /** store prism value of info */
    private InfoPrism _info;

    /** store prism value of tags */
    private List<TagPrism> _tags;

    public static final String PRISM_TYPE = "io.swagger.v3.oas.annotations.OpenAPIDefinition";

```

### CODE REVIEW

File: src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module io.avaje.prism {
+
+  exports io.avaje.prism;
+  requires java.compiler;
+
```
Comment (rbygrave) [inline src/main/java/module-info.java:5]: I'm thinking we need to add the `provides javax.annotation.processing.Processor with ... `

Comment (rbygrave) [reply src/main/java/module-info.java:5]: I'll merge this PR and add that in I reckon ...

Comment (SentryMan) [reply src/main/java/module-info.java:5]: ah yeah forgot about that

**Comment by rbygrave** [comment]
Do you see the error below running tests?
```
[ERROR] io.avaje.prisms.test.PrismProcessorTest.runAnnotationProcessor  Time elapsed: 0.751 s  <<< ERROR!
java.lang.RuntimeException: java.lang.NullPointerException: Cannot invoke "javax.lang.model.type.DeclaredType.asElement()" because the return value of "io.avaje.prism.internal.GeneratePrismPrism.value()" is null

```

**Comment by SentryMan** [comment]
yeah lemme fix that

**Comment by SentryMan** [comment]
fixed

**Comment by SentryMan** [comment]
ok now we're good

**Pull request merged**